### ISSUE TITLE
adding to an exif 

### ISSUE THREAD

**Issue by MikeSavad** [easydiffusion/easydiffusion#1614]
Every image I make I have to copy down the info for. And while I know you have a txt file in the backup. I'd like to be able to get all the parameters installed into the EXIF in the jpgs for each file so I can just download it and use it. Instead of copying, editing etc.

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by cmdr2** [comment]
Hi @MikeSavad Have you tried setting "Metadata Format" (in the Settings tab) to `Embed`?

That'll save the metadata into the JPEG file, in the comments field.

For e.g. 
![image](https://github.com/easydiffusion/easydiffusion/assets/844287/7fce84d7-5c6f-4b6d-92d4-77972812560b)

Please let me know if you meant something else. Thanks!

**Comment by MikeSavad** [comment]
Little easter eggs hidden in the settings. I'll give that a try, thanks. This would be added to the downloaded one or just the backup?

**Comment by woaiseeit** [comment]
It's added only to the automatically saved one, not to the downloaded image. (Probably that's why it can be enabled only when "Auto-Save Images" is enabled. )

FWIW, the data looks like this:
```
Image:
  [...]
  Properties:
    clip_skip: False
    date:create: 2023-10-08T01:36:15+00:00
    date:modify: 2023-10-08T01:32:00+00:00
    date:timestamp: 2023-10-08T01:36:53+00:00
    guidance_scale: 3.0
    negative_prompt: XXX
    num_inference_steps: 30
    png:IHDR.bit-depth-orig: 8
    png:IHDR.bit_depth: 8
    png:IHDR.color-type-orig: 2
    png:IHDR.color_type: 2 (Truecolor)
    png:IHDR.interlace_method: 0 (Not interlaced)
    png:IHDR.width,height: 512, 768
    png:text: 17 tEXt/zTXt/iTXt chunks were found
    png:width: 512
    prompt: XXX
    sampler_name: ddim
    seed: 138126055
    signature: XXX
    tiling: None
    use_controlnet_model: None
    use_embeddings_model: None
    use_face_correction: None
    use_lora_model: None
    use_stable_diffusion_model: deliberate_v2
    use_upscale: None
    use_vae_model: vae-ft-mse-840000-ema-pruned
```

**Comment by MikeSavad** [comment]
Would it be possible to have it added to the downloaded one? The auto is just a save as. Though I did try it and It didn't seem to record it.  I would want that stuff and the phrasing.  Right now its the only way to keep track of things.  And when I copy it, I end up getting template info from the other boxes it seems to copy.

**Comment by Martyprod** [comment]
> Hi @MikeSavad Have you tried setting "Metadata Format" (in the Settings tab) to `Embed`?
> 
> That'll save the metadata into the JPEG file, in the comments field.
> 
> For e.g. ![image](https://user-images.githubusercontent.com/844287/271779436-7fce84d7-5c6f-4b6d-92d4-77972812560b.png)
> 
> Please let me know if you meant something else. Thanks!


I tried to use these metadata in other software to better catalog them or just read them but it's not Exif, so it can't be read by picture viewer (like Faststone). Would be great if there was some possibilities to access or read these metadata.
I tried several windows column to display theme but it fails, only way to display them, is right click on the file, property, detalis, and read them on the comments part. i certainly miss something, ? thanks !

**Comment by MikeSavad** [comment]
Mostly I want to keep everything I did in the file itself should I want to try repeating it again. And the only way is to just copy and paste. I use Xnview - Ctrl I  and I can see the exif - meta stuff there. I guess its not really called an exif that's a camera term. Its just a comments section to write down stuff.  I don't think I ever used the ratings thing in any program unless it was an accident.


----Mike Savad

**Comment by Martyprod** [comment]
> Mostly I want to keep everything I did in the file itself should I want to try repeating it again. And the only way is to just copy and paste. I use Xnview - Ctrl I and I can see the exif - meta stuff there. I guess its not really called an exif that's a camera term. Its just a comments section to write down stuff. I don't think I ever used the ratings thing in any program unless it was an accident.
> 
> ----Mike Savad

hi sorry to bother you, i just installed Xnview, hit Control + I, but don't see the ED metadata there, how did you do it ? thanks !

**Comment by MikeSavad** [comment]
Make sure its a jpg, on the image, then it should just load up, then it will show itpc. You can right click and edit that too.
 
----Mike Savad

**Comment by MikeSavad** [comment]
It would be nice to save it. Right now it looks like this in the comment:

  Remove Use these settings Save as templateA cat's head made from transparent, highly refractive cut crystal, surreal

Seed: 3182617, Dimensions: 3264x2112, Sampler: unipc_tu, Inference Steps: 100, Guidance Scale: 7.5, Model: NEW acornIsSpinning_acornxlV1, Negative Prompt: Black and white, floating objects, low quality, wrong anatomy, logo, signature, floating object, disconnected objects, monochrome, watermark, bad crops, poor cropping, extra legs, extra arms, wrong spelling, missing legs, missing feet, worst quality, low quality, jpeg artifacts, snapshot, low resolution, low details,lights in bottles, Prompt Strength: 0.2
Processed 1 images in 7 minutes

which, its all there, but its one big block of messy text and hard to find what I want if I just want a description. 


----Mike Savad

**Comment by Martyprod** [comment]
> which, its all there, but its one big block of messy text and hard to find what I want if I just want a description.

yes, we search the same solution. one of my friend is a dev. i'm sure he can do a small python app to display each stuff correctly, as it's eparated by by , and space, with the title of the information with a :  , 
like
Seed: 3182617
Dimensions: 3264x2112
Sampler: unipc_tu
Inference Steps: 100

etc.. that i would really love, is having this information displayed separatly in a picture viewer (or even better in a file browser with picture display like directory opus. and being able to filer by sampler, model, etc ... it's curious that nobody did such a small software. to have a database of all the rendering done. because yes, it's actually a big block mess text, difficul to use !
